mlflow/store/artifact/cloud_artifact_repo.py

```diff
@@ -0,0 +1,252 @@
+        """
+        pass
+
+    @abstractmethod
+    def _upload_to_cloud(self, cloud_credential_info, src_file_path, artifact_file_path):
+        """
+        Upload a single file to the cloud.
+        :param cloud_credential_info: ArtifactCredentialInfo containing presigned URL for the
+                                      current file
+        :param src_file_path: source file path
+        :param artifact_file_path: artifact path, including file name
+        :return:
+        """
+        pass
+
+    # Read APIs
+
+    def _extract_headers_from_credentials(self, headers):
+        """
+        :return: A python dictionary of http headers converted from the protobuf credentials
+        """
+        return {header.name: header.value for header in headers}
+
+    def _parallelized_download_from_cloud(self, file_size, remote_file_path, local_path):
+        read_credentials = self._get_read_credential_infos([remote_file_path])
+        # Read credentials for only one file were requested. So we expected only one value in
+        # the response.
+        assert len(read_credentials) == 1
+        cloud_credential_info = read_credentials[0]
+        # from mlflow.utils.databricks_utils import get_databricks_env_vars
+
+        try:
+            parallel_download_subproc_env = os.environ.copy()
+            # parallel_download_subproc_env.update(
+            #     get_databricks_env_vars(self.databricks_profile_uri)
+            # )
+            failed_downloads = parallelized_download_file_using_http_uri(
+                thread_pool_executor=self.chunk_thread_pool,
+                http_uri=cloud_credential_info.signed_uri,
+                download_path=local_path,
+                file_size=file_size,
+                uri_type=cloud_credential_info.type,
+                chunk_size=_DOWNLOAD_CHUNK_SIZE,
+                env=parallel_download_subproc_env,
+                headers=self._extract_headers_from_credentials(cloud_credential_info.headers),
+            )
+            download_errors = [
+                e for e in failed_downloads.values() if e["error_status_code"] not in (401, 403)
+            ]
+            if download_errors:
+                raise MlflowException(
+                    f"Failed to download artifact {remote_file_path}: " f"{download_errors}"
+                )
+
+            if failed_downloads:
+                new_cloud_creds = self._get_read_credential_infos([remote_file_path])[0]
+                new_signed_uri = new_cloud_creds.signed_uri
+                new_headers = self._extract_headers_from_credentials(new_cloud_creds.headers)
+
+                for i in failed_downloads:
+                    download_chunk(i, _DOWNLOAD_CHUNK_SIZE, new_headers, local_path, new_signed_uri)
+        except Exception as err:
+            if os.path.exists(local_path):
+                os.remove(local_path)
+            raise MlflowException(err)
+
+    def _download_file(self, remote_file_path, local_path):
```

Comment:
Compared to the original `_download_file` in `DatabricksArtifactRepository`, much like in the write case (explained in [this comment](https://github.com/mlflow/mlflow/pull/9444/files#r1305122245)), I've removed this run relative path joining logic:
```
run_relative_remote_file_path = posixpath.join(
            self.run_relative_artifact_repo_root_path, remote_file_path
        )
```
because other subclasses of `CloudArtifactRepository` don't have `self.run_relative_artifact_repo_root_path`. Instead, I've moved it into the `_get_read_credential_infos` implementation of `DatabricksArtifactRepository`, which was the only place the run relative path was used for download anyways.